webrender/res/ps_box_shadow.fs.glsl

```diff
@@ -3,7 +3,18 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 void main(void) {
-    vec2 uv = min(vec2(1.0), vMirrorPoint - abs(vUv.xy - vMirrorPoint));
+    // Mirror and stretch the box shadow corner over the entire
+    // primitives.
+    vec2 uv = vMirrorPoint - abs(vUv.xy - vMirrorPoint);
+
+    // Ensure that we don't fetch texels outside the box
+    // shadow corner. This can happen, for example, when
+    // drawing the outer parts of an inset box shadow.
+    uv = clamp(uv, vec2(0.0), vec2(1.0));
+
+    // Map the unit UV to the actual UV rect in the cache.
     uv = mix(vCacheUvRectCoords.xy, vCacheUvRectCoords.zw, uv);
```

Comment:
I don't think that's going to be enough. Given that we are doing a regular `texture` call for sampling, and the sampler is bilinear, and `vCacheUvRectCoords` denotes the bounds of the UV space, sampling on that bound (when uv==0 or uv==1) would touch the neighboring texels. If that sounds reasonable to you, we should subtract half a texel from the bounds.